### PR TITLE
Minor rewrite of char primitive unicode intro.

### PR DESCRIPTION
**Pull request by frewsxcv**
Opened primarily to address #36998.

Despite my love for emoji, the heart example is a little confusing because both heart characters start with the same code point and there can be stark rendering differences across browsers. I also spelled out what each of the code points is in the code block, which (hopefully) sheds light why one character is one code point while the other is two.

Very much open to suggestion and improvements. I'm pretty tired when I wrote this so I might wake up and realize that this is making things more confusing 😅

### CODE REVIEW

File: src/libstd/primitive_docs.rs [hunk 1]
```diff
@@ -103,26 +103,31 @@ mod prim_bool { }
...
+/// // U+0065: 'latin small letter e'
+/// assert_eq!(Some('\u{0065}'), chars.next());
+/// // U+0301: 'combining acute accent'
+/// assert_eq!(Some('\u{0301}'), chars.next());
+/// assert_eq!(None, chars.next());
 /// ```
 ///
-/// This means it won't fit into a `char`. Trying to create a literal with
-/// `let heart = '❤️';` gives an error:
+/// This means that the contents of the first string above _will_ fit into a
```
Comment (frewsxcv) [inline src/libstd/primitive_docs.rs]: Note to self: it's not a string technically, it's a string slice, but that might be getting too deep in the weeds

Comment (QuietMisdreavus) [inline src/libstd/primitive_docs.rs]: Eh, i often use lowercase "string" to refer to text data in general, and uppercase "String" (potentially with backticks) to denote the owned-text-data type in the standard library. I don't think it's a problem here.

**Comment by QuietMisdreavus** [comment]
@bors r+ rollup

Thanks! I like this example better, especially since you put in the character names in there.

**Pull request merged**